stdlib/public/core/StringCreate.swift

```diff
@@ -213,7 +212,34 @@ extension String {
     repair: Bool
   ) -> (String, repairsMade: Bool)?
   where Input.Element == Encoding.CodeUnit {
-    // TODO(String Performance): Attempt to form smol strings
+    if input.count < _SmallString.capacity {
+      var repaired = false
+      var overflow = false
+      let result = _SmallString(initializingUTF8With: { buffer in
+        var bytesUsed = 0
+        repaired = transcode(
+          input.makeIterator(),
+          from: encoding,
+          to: UTF8.self,
+          stoppingOnError: false,
+          into: {
+            if bytesUsed < buffer.count {
+              buffer[bytesUsed] = $0
+            }
+            bytesUsed &+= 1
+          }
+        )
+        guard bytesUsed <= buffer.count else {
+          overflow = true
+          return 0
```

Comment:
I wonder if you can save the transcoding already done instead of having to start over for the large case. It would be better if `transcode()` took its iterator `inout` though.

Reply:
My hope is that transcoding 14 bytes is cheap enough to not be a big deal, but yeah, would be nice

Reply:
It's probably fine, yes. On thinking about it more, this is exactly why I ended up re-implementing the transcoding loop in `_validate()`. There might be an opportunity to share some implementation between these cases at some later time.